src/app/clusters/service-area-server/service-area-server.cpp

```diff
@@ -0,0 +1,1104 @@
+        {
+            exitResponse(SelectLocationsStatus::kInvalidSet, "decoding of newLocations parameter size failed"_span);
+            return;
+        }
+
+        // If the device determines that it can't operate at all locations from the list,
+        // the SelectLocationsResponse command's Status field SHALL indicate InvalidSet.
+        if (((numberOfLocations == 0) || (numberOfLocations > kMaxNumSelectedLocations)))
+        {
+            exitResponse(SelectLocationsStatus::kInvalidSet, "invalid number of locations"_span);
+            return;
+        }
+    }
+
+    // if number of selected locations in parameter matches number in attribute - the locations *might* be the same
+    matchesCurrentSelectedLocations = (numberOfLocations == mDelegate->GetNumberOfSelectedLocations());
+
+    if (!req.newLocations.IsNull())
+    {
+        // do as much parameter validation as we can
+        {
+            uint32_t i         = 0;
+            auto iLocationIter = req.newLocations.Value().begin();
+            while (iLocationIter.Next())
+            {
+                uint32_t aSelectedLocation = iLocationIter.GetValue();
+
+                // each item in this list SHALL match the LocationID field of an entry on the SupportedLocations attribute's list
+                // If the Status field is set to UnsupportedLocation, the StatusText field SHALL be an empty string.
+                if (!IsSupportedLocation(aSelectedLocation))
+                {
+                    exitResponse(SelectLocationsStatus::kUnsupportedLocation, ""_span);
+                    return;
+                }
+
+                // Checking for duplicate locations.
+                uint32_t j         = 0;
+                auto jLocationIter = req.newLocations.Value().begin();
+                while (j < i)
+                {
+                    jLocationIter.Next();
+                    if (jLocationIter.GetValue() == aSelectedLocation)
+                    {
+                        exitResponse(SelectLocationsStatus::kDuplicatedLocations, ""_span);
+                        return;
+                    }
+                    j += 1;
+                }
+
+                // check to see if parameter list and attribute still match
+                if (matchesCurrentSelectedLocations)
+                {
+                    if (!mDelegate->GetSelectedLocationByIndex(listIndex, oldSelectedLocation) ||
+                        (aSelectedLocation != oldSelectedLocation))
+                    {
+                        matchesCurrentSelectedLocations = false;
+                    }
+                }
+
+                i += 1;
+            }
+
+            // after iterating with Next through DecodableType - check for failure
+            if (CHIP_NO_ERROR != iLocationIter.GetStatus())
+            {
+                exitResponse(SelectLocationsStatus::kInvalidSet, "decoding of newLocations parameter failed"_span);
```

Comment:
No, malformed TLV should be INVALID_COMMAND.

Reply:
fixed [here](https://github.com/project-chip/connectedhomeip/pull/33991/commits/c8358f1e8f8d38af9dbf71135f96d99406791858).